src/composables/PluginManager.js

```diff
@@ -0,0 +1,149 @@
+import { readTextFile } from 'src/composables/Files';
+import plugins from 'src/plugins';
+
+let instanciatePlugins = [];
+
+/**
+ * Initialize plugin metadata and definitions.
+ *
+ * @param {Plugin} plugin - instantiate plugin.
+ */
+export function initPluginMetadata(plugin) {
+  plugin.metadata = new plugin.pluginModel.PluginMetadata();
+
+  const definitions = plugin.metadata.getDefinitions();
+
+  // Add url file in svgImage for componentDefinitionCard
+  definitions.components.forEach((def) => {
+    if (def.icon) {
+      def.svgImage = `/plugins/${plugin.name}/icons/${def.icon}.svg`;
+    }
+  });
+
+  plugin.definitions = definitions;
+}
+
+/**
+ * Retreive files' informations.
+ *
+ * @param {Plugin} plugin - instantiate plugin.
+ * @param {String} defType - Type of file to retreive.
+ * @return {Array<Object>} Promise with files' information on success otherwise an error.
+ */
+export function getFilesInfo(plugin, defType) {
+  return plugin.definitions[defType].reduce((acc, def) => {
+    if (def.model) {
+      acc.push({
+        name: def.model,
+        type: 'models',
+        path: `/plugins/${plugin.name}/models/${def.model}.svg`,
+      });
+    }
+
+    if (def.icon) {
+      acc.push({
+        name: def.icon,
+        type: 'icons',
+        path: `/plugins/${plugin.name}/icons/${def.icon}.svg`,
+      });
+    }
+
+    return acc;
+  }, []);
+}
+
+// Remove if svg import is possible
+/**
+ * Create plugin resources.
+ *
+ * @param {Plugin} plugin - instantiate plugin
+ * @return {Promise<Object>} Promise with resources on success otherwise an error.
+ */
+export async function createPluginResources(plugin) {
+  const files = [...new Set(getFilesInfo(plugin, 'components'))];
```

Comment:
can you move this to `getFilesInfo` instead, and make `getFilesInfo` retrieve all unique files :) ?

Reply:
done